src/trans.rs

```diff
@@ -1310,25 +1337,12 @@ impl<'v, 'tcx: 'v, 'module: 'v> BinaryenFnCtxt<'v, 'tcx, 'module> {
     }
 
     fn trans_lval(&mut self, lvalue: &Lvalue<'tcx>) -> Option<BinaryenLvalue> {
+        let mir = self.mir.borrow();
+
         let i = match *lvalue {
-            Lvalue::Arg(i) => i.index() as u32,
-            Lvalue::Var(i) => {
+            Lvalue::Local(i) => {
```

Comment:
You should be able to use the index uniformly now (the flattening I suggested to @brson a long time ago is what we always do nowadays in rustc itself).